src/components/Chat/InputMessageV2.tsx

```diff
@@ -375,18 +377,26 @@ const InputMessageV2 = ({
       return;
     }
 
-    Alert.alert(
-      '',
-      type === ANONYMOUS
-        ? `Switch back to regular chat?\nMessage ${username} using your username.`
-        : 'Switch to anonymous chat?\nMessage this user anonymously instead.',
-      [
-        {
-          text: 'Cancel'
-        },
-        {text: 'Yes, move to other chat', onPress: onToggleConfirm}
-      ]
-    );
+    if (!isAnonimityEnabled) {
+      ToastMessage.show({
+        type: 'asNative',
+        text1: 'This user does not want to receive anonymous messages',
+        position: 'bottom'
+      });
+    } else {
+      Alert.alert(
+        '',
+        type === ANONYMOUS
+          ? `Switch back to regular chat?\nMessage ${username} using your username.`
+          : 'Switch to anonymous chat?\nMessage this user anonymously instead.',
+        [
+          {
+            text: 'Cancel'
+          },
+          {text: 'Yes, move to other chat', onPress: onToggleConfirm}
+        ]
+      );
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This new block of code introduces a conditional check for `isAnonimityEnabled`. If it's `false`, a toast message is shown, otherwise an alert is displayed. The logic seems fine, but consider extracting this into a separate function for better readability and maintainability.



<!-- This is an auto-generated comment by OSS CodeRabbit -->